src/declarativedbusinterface.cpp

```diff
@@ -377,7 +377,10 @@ QVariant DeclarativeDBusInterface::getProperty(const QString &name)
 
     QVariant v = reply.arguments().first();
     if (v.userType() == qMetaTypeId<QDBusVariant>())
-        return v.value<QDBusVariant>().variant();
+        if (v.value<QDBusVariant>().variant().userType() == qMetaTypeId<QDBusArgument>())
```

Comment:
The logic looks okay, but could you add braces to the outer if/else block.  And rather than repeating v.value<QDBusVariant>().variant() I'd suggest assigning the result of that to v as  a first step and use that for the rest of the logic.